### DISCUSSION TITLE
Delay on attiny-hal fails with no method in scope

### DISCUSSION THREAD

**Discussion by flpydsk** [Rahix/avr-hal#310] [Q&A] [answered]
using rev e897783816437a677aa577ddfdaa34e9a1e86d96
attiny-hal with an attiny85
delay does not work. failed with compiler error:
```text
error[E0599]: no method named `delay_ms` found for struct `attiny_hal::delay::Delay<attiny_hal::clock::MHz1>` in the current scope
```

reproduce with:

```rust
#![no_std]
#![no_main]

use attiny_hal::delay::*;
use attiny_hal::clock::*;
extern crate panic_halt;
#[attiny_hal::entry]
fn main() -> ! {
    let pins = attiny_hal::pins!(attiny_hal::Peripherals::take().unwrap());
    let mut delay = Delay::<MHz1>::new();
    let mut led = pins.pb4.into_output();
    loop {
        led.toggle();
        delay.delay_ms(1000);
    }

}
```

**Comment by Rahix** [comment, marked as answer]
The full error message already has the hint to fix this :)

<pre><font color="#F92672"><b>error[E0599]</b></font><b>: no method named `delay_ms` found for struct `attiny_hal::delay::Delay&lt;attiny_hal::clock::MHz1&gt;` in the current scope</b>
  <font color="#66D9EF"><b>--&gt; </b></font>examples/trinket/src/bin/trinket-blink.rs:16:15
   <font color="#66D9EF"><b>|</b></font>
<font color="#66D9EF"><b>16</b></font> <font color="#66D9EF"><b>| </b></font>        delay.delay_ms(1000);
   <font color="#66D9EF"><b>| </b></font>              <font color="#F92672"><b>^^^^^^^^</b></font> <font color="#F92672"><b>method not found in `attiny_hal::delay::Delay&lt;attiny_hal::clock::MHz1&gt;`</b></font>
   <font color="#66D9EF"><b>|</b></font>
   <font color="#66D9EF"><b>= </b></font><b>help</b>: items from traits can only be used if the trait is in scope
<font color="#A1EFE4"><b>help</b></font>: the following trait is implemented but not in scope; perhaps add a `use` for it:
   <font color="#66D9EF"><b>|</b></font>
<font color="#66D9EF"><b>4</b></font>  <font color="#66D9EF"><b>| </b></font>use embedded_hal::blocking::delay::DelayMs;
   <font color="#66D9EF"><b>|</b></font>
</pre>

Just add

```rust
use embedded_hal::blocking::delay::DelayMs;
```

to your imports.  You'll also need to explicitly specify the type of the delay integer:

```rust
        delay.delay_ms(1000u16);
```
